src/hooks/useMessageList.js

```diff
@@ -0,0 +1,219 @@
+  } catch (error) {
+    throw new Error('Failed to update ttl file.');
+  }
+};
+
+export { updateMessageReadStatus };
+
+/**
+ * @typedef {object} messageListHookObject
+ * @property {boolean} isLoading - if the message list is loading
+ * @property {boolean} isSuccess - if the message list has been loaded
+ * @property {boolean} isError - if an error occurred while fetching the message
+ * list
+ * @property {object} error - the error that occurred while fetching
+ * @property {Array} data - the list of messageObjects containing the message
+ * content and information
+ * @property {Function} refetch - function to refetch data
+ * @property {boolean} isFetching - is the collection being fetched
+ * @property {(messageObject) => Promise} add - updates the messageObject from messageListObject
+ */
+
+/**
+ * Custom hook that helps renders messages from Inbox/Outbox from PASS container
+ * in Solid. It's a wrapper for a React Query and manages the messageListObject
+ * pulled and parsed from Solid
+ *
+ * @memberof hooks
+ * @function useMessageList
+ * @param {string} messageType - The string for 'Inbox' or 'Outbox'
+ * @returns {messageListHookObject} messageListObject - all the data provided by
+ * the useQuery call
+ */
+const useMessageList = (messageType) => {
+  const { session, podUrl } = useSession();
+  const { fetch } = session;
+  const [storedDataset, setStoredDataset] = useState(null);
+  const queryClient = useQueryClient();
+  const containerUrl = podUrl && new URL(`PASS/${messageType}/`, podUrl).toString();
+  let messageList = [];
+
+  const parse = async (data) => {
+    const solidDataset = getThingAll(data);
+    const allMessagesDataset = solidDataset.filter((thing) => thing.url.endsWith('ttl'));
+
+    try {
+      messageList = await Promise.all(
+        allMessagesDataset.map(async (message) => {
+          const messageDataset = await getSolidDataset(message.url, { fetch });
+          const messageThing = getThingAll(messageDataset);
+          return parseMessageTTL(messageThing);
+        })
+      );
+    } catch {
+      return [];
+    }
+
+    return messageList?.sort((a, b) => b.uploadDate - a.uploadDate);
+  };
+
+  const updateMessageList = async (messageObject) => {
+    await updateMessageReadStatus(messageObject, fetch);
+  };
+
+  const fetchDocument = async () => {
+    let myDataset;
+    try {
+      myDataset = await getSolidDataset(containerUrl.toString(), { fetch });
+    } catch (e) {
+      if (e.response.status === 404) {
+        myDataset = createSolidDataset();
+        myDataset = await saveSolidDatasetAt(containerUrl.toString(), myDataset, { fetch });
+      } else {
+        throw e;
+      }
+    }
+    setStoredDataset(myDataset);
+    return parse(myDataset);
+  };
+
+  const { isLoading, isError, error, data, isSuccess, refetch, isFetching } = useQuery({
+    queryKey: [containerUrl?.toString()],
+    queryFn: fetchDocument
+  });
+
+  const addMutation = useMutation({
```

Comment:
Had custom `fetchDocument` and `addMutation` functions from `usRdfCollection.js` included for this use case.

Reply:
We can probably make some enhancements to `useRdfCollection` to better support this situation. But that can wait for a followup PR.

Reply:
Got it, think that would be great.

Reply:
Reading up on https://tanstack.com/query/v3/docs/react/guides/updates-from-mutation-responses